modules/generic_worker/templates/run-generic-worker.sh.erb

```diff
@@ -10,10 +10,36 @@
 # to numberOfTasksToRun, the worker won't exit and machine won't reboot.
 rm -f tasks-resolved-count.txt
 
+source /usr/local/bin/bugs.sh
+
 # First run the generic-worker, passing through any arguments handed to this
 # wrapper script...
 /usr/local/bin/generic-worker "$@" 2>&1 | logger -t generic-worker -s
 
+exitstatus=( ${PIPESTATUS[0]} )
+
+export TASKCLUSTER_CLIENT_ID="<%= @quarantine_client_id %>"
+export TASKCLUSTER_ACCESS_TOKEN="<%= @quarantine_access_token %>"
+
+if [ $exitstatus -eq 69 ]; then
+    if [ -f /Users/cltbld/quarantined ]; then
+        echo "Worker is quarantined"
```

Comment:
If `/Users/cltbld/quarantined` exists, it means on the previous run, the worker was quarantined. When this happens at this point in the code, we know the worker believes itself to be in an invalid bad/corrupted state after this new run. Therefore, this could be because either:

1) The failure mode is hit without claiming a task, e.g. on generic-worker initialisation. In this case, the machine will start boot-looping - that could be potentially harmful to the hardware if it is e.g. rebooting every minute or so. If several machines (e.g. potentially an entire worker pool) start doing this at the same time due to a code bug, I'm concerned this could be even more damaging, even to other machines in the data centre.

2) The worker quarantine was lifted or expired naturally, the queue provided a task for the worker, and during task execution the failure mode was discovered. In this case, we are not in a boot-looping situation, since the worker will be quarantined again, and will hit an idle timeout on the next run if the quarantine is not lifted, which occurs after 96 hours.

The long and short of this is that I think it would be wise to set a minimum worker uptime duration between worker reboots in the case when exit code 69 is detected. If a worker fails immediately with an error code 69, we shouldn't immediately reboot. There is a further argument to say once we hit exit code 69, the worker should just avoid the reboot entirely, and leave the machine running. The only reason to keep rerunning generic-worker would be to catch the case where the root cause is of the problem gets resolved and the quarantine is actively lifted, and we want the worker to immediately start taking tasks again. In this case, it removes the need for someone to reboot all affected workers.

A happy middle ground may be to limit rebooting to once per hour in the case of an exit code 69. Note, this restriction shouldn't apply for other exit codes, since e.g. if a worker runs a task that only lasts one minute, it should still reboot immediately.